### PR TITLE
Generator cleanup with tests

### PR DESCRIPTION
**Pull request by mrhardikjoshi**
This PR introduces tests on top of changes of #1716

### CODE REVIEW

File: e2e_tests/setup.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+fi
+
+if [[ -n "$SHA" ]]
+then
+  sed 's/.*gem '"'"'activegraph'"'"'.*/gem '"'"'activegraph'"'"', github: "neo4jrb\/activegraph", ref: "'"$SHA"'"/' docs/activegraph.rb > template.tmp
+else
+  echo "SHA=$(git rev-parse "$GITHUB_SHA")" >> $GITHUB_ENV
+  sed 's/.*gem '"'"'activegraph'"'"'.*/gem '"'"'activegraph'"'"', github: "neo4jrb\/activegraph", ref: "'"$(git rev-parse "$GITHUB_SHA")"'"/' docs/activegraph.rb > template.tmp
+fi
+rails _7.1.2_ new myapp -O -m ./template.tmp
```
Comment (mrhardikjoshi) [inline e2e_tests/setup.sh]: make rails version dynamic

File: e2e_tests/setup.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+if [[ -n "$ACTIVE_MODEL_VERSION" ]]
+then
+  gem install rails -v $ACTIVE_MODEL_VERSION --no-document
+else
+  gem install rails -v 7.1.2 --no-document
+fi
+
+if [[ -n "$SHA" ]]
```
Comment (mrhardikjoshi) [inline e2e_tests/setup.sh]: option to pick path from environment variable

**Comment by mrhardikjoshi** [comment]
update readme file to include command/steps to run e2e tests

File: docs/activegraph.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
 environment generator
 environment nil, env: 'development' do <<END
-config.neo4j.driver.url = 'neo4j://localhost:7472'
+config.neo4j.driver.url = 'bolt://localhost:7687'
```
Comment (klobuczek) [inline docs/activegraph.rb]: the port was unconventional but the protocol was right

**Review by klobuczek — DISMISSED** [review]
Great job!

**Comment by mrhardikjoshi** [comment]
To Do Notes:
1) add tests for down migration command 
2) add tests for running default rails tests in minitest
3) cleanup and merge both github workflows

**Review by klobuczek — APPROVED** [review]

**Pull request merged**